src/extension.ts

```diff
@@ -105,13 +105,17 @@ function getClientConfig(context: ExtensionContext) {
     ['diagnostics.onParse', 'diagnostics.onParse'],
     ['diagnostics.onType', 'diagnostics.onType'],
     ['codeLens.localVariables', 'codeLens.onLocalVariables'],
+    ['formatting.enabled', 'formatting.enabled'],
```

Comment:
If you wire this preference up into cquery, then cquery will not announce it is a formatting provider, which will remove the need for the middleware.

Reply:
Done.